README.md

```diff
@@ -46,49 +318,95 @@ dtypes = {
     "SibSp": pl.Categorical,
     "Parch": pl.Categorical
 }
-df = pl.read_csv("titanic.csv", dtype=dtypes)
-```
-
-From the polars dataframe, we create a metadataset and store it in a JSON file that follows the GMF standard:
-
-```python
 
-dataset = MetaDataset.from_dataframe(df)
-dataset.to_json("test.json")
+df = pl.read_csv(dataset_csv, dtypes=dtypes)
 ```
 
-## Note on pandas
-
+<details>
+     <summary> 
+     Note on using Pandas
+     </summary>
+     
 Internally, MetaSynth uses polars (instead of pandas) mainly because typing and the handling of non-existing data is more
 consistent. It is possible to supply a pandas DataFrame instead of a polars DataFrame to `MetaDataset.from_dataframe`.
 However, this uses the automatic polars conversion functionality, which for some edge cases result in problems. Therefore,
 we advise users to create polars DataFrames. The resulting synthetic dataset is always a polars dataframe, but this can
 be easily converted back to a pandas DataFrame by using `df_pandas = df_polars.to_pandas()`.
+</details>
 
+##### 2. Next, we can generate a metadataset from the polars dataframe.
 
-<!-- CONTRIBUTING -->
+```python
+# create metadata
+metadata = ms.MetaDataset.from_dataframe(df)
+```
 
-## Contributing
+> Note: if at this point you get the following warning about a potential unique variable, do not worry, it is safe to continue.
+> 
+> ```
+> Variable PassengerId seems unique, but not set to be unique. Set the variable to be either unique or not unique to remove this warning. warnings.warn(f"\nVariable {series.name} seems unique, but not set to be unique.\n"
+> ```
+
+##### 3. We can export this metadataset to a .JSON file using:
 
+```python
+#export metadata
+metadata.to_json("metadata.json")
+```
+
+> Note: if at this point you get the following warnings, do not worry, it is safe to continue.
+> 
+> ```
+> DeprecationWarning: read_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice. 
+> DeprecationWarning: open_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
+> ```
+
+
+
+#### Generating synthetic data
+
+##### 1. We can load metadata from a .JSON file:
+```python
+# load metadata
+metadata = ms.MetaDataset.from_json("metadata.json")
+```
+
+> Note: if at this point you get the following warnings, do not worry, it is safe to continue.
+> 
+> ```
+> DeprecationWarning: read_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice. 
+> DeprecationWarning: open_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
```

Comment:
Agreed, for now I think it's good to keep the error warnings in. But, I'll try resolving these issues in the near future so that the warnings can be removed :)

Reply:
Yes, it will be fixed one way or the other, so you can just remove it in this PR and assume the warning doesn't get thrown.